### PR TITLE
fix: replace useBinder with useForm

### PR DESCRIPTION
**Pull request by taefi**
Also, replace `useBinderNode` with `useFormPart`.
Also, take out binding-arrays.adoc article from React form guides.

### CODE REVIEW

**Comment by Lodin** [comment]
Actually, it should be `useFormState`, not `...Context` as we concluded in vaadin/hilla#1186

File: articles/lit/guides/forms/binder-validation.adoc [hunk 1]
```diff
@@ -459,31 +459,31 @@ export default function ProfileView() {
 
   const { model, field, addValidator } = useForm(PersonModel);
 
-  const nameBinderNode = useFormPart(model.password);
-  const usernameBinderNode = useFormPart(model.username);
-  const ageBinderNode = useFormPart(model.age);
-  const emailBinderNode = useFormPart(model.email);
+  const nameField = useFormPart(model.password);
```
Comment (sissbruecker) [inline articles/lit/guides/forms/binder-validation.adoc:462]:
```suggestion
  const nameField = useFormPart(model.name);
```

File: articles/lit/guides/forms/form-status-tracking.adoc [hunk 2]
```diff
@@ -39,7 +39,31 @@ endif::[]
 
 The following properties are available both for the form as a whole, and for each field independently.
 They are a part of the <<reference#binder-node,`BinderNode`>> interface.
+ifdef::react[]
+Assume you have the [classname]`UseFormResult` instance by calling [methodname]`useForm`, then you can access form status properties as:
```
Comment (sissbruecker) [inline articles/lit/guides/forms/form-status-tracking.adoc:43]:
```suggestion
You can access form status properties for the whole form from the [classname]`UseFormResult` instance returned from [methodname]`useForm`:
```
Thanks for adding the distinction for React BTW 👍

File: articles/lit/guides/forms/form-status-tracking.adoc [hunk 3]
```diff
@@ -39,7 +39,31 @@ endif::[]
...
+[source,tsx]
+----
+const form = useForm(EntityModel);
+
+const invalid = form.invalid;
+const dirty = form.dirty;
+const errors = form.errors;
+...
+----
+and the same goes for a single field when the [classname]`UseFormResult` instance obtained by calling [methodname]`useForm`, then you can access these properties as:
```
Comment (sissbruecker) [inline articles/lit/guides/forms/form-status-tracking.adoc:53]:
```suggestion
To access form status properties for individual fields, use the [classname]`UseFormPartResult` instance returned from [methodname]`useFormPart`:
```

File: articles/lit/guides/forms/form-status-tracking.adoc [hunk 4]
```diff
@@ -39,7 +39,31 @@ endif::[]
...
+[source,tsx]
+----
+const form = useForm(EntityModel);
+const nameField = useFormPart(form.mode.name);
```
Comment (sissbruecker) [inline articles/lit/guides/forms/form-status-tracking.adoc:57]:
```suggestion
const nameField = useFormPart(form.model.name);
```

File: articles/lit/guides/forms/binder-load.adoc [hunk 5]
```diff
@@ -19,13 +19,13 @@ You can use the [methodname]`read()` method from the [classname]`BinderControl`
 [source,tsx]
```
Comment (sissbruecker) [inline articles/lit/guides/forms/binder-load.adoc:19]: The text right above still references `BinderControl`.

Comment (taefi) [reply articles/lit/guides/forms/binder-load.adoc:19]: Nice catch! Done!

**Review by sissbruecker — APPROVED** [review]

**Pull request merged**